### PR TITLE
Null safety

### LINKED ISSUE #10: [Feature Request] Add sound null safety support
Please add sound null safety support.

### LINKED ISSUE #11: Don't use outdated libraries
The logging version referenced is outdated, which results in errors like this:

```
Running "flutter pub get" in roji_wallet...                     
Because no versions of stream_chat_flutter_core match >2.2.1 <3.0.0 and stream_chat_flutter 2.2.1 depends on stream_chat_flutter_core ^2.2.1, stream_chat_flutter 2.2.1 requires stream_chat_flutter_core 2.2.1.
And because stream_chat_flutter_core 2.2.1 depends on stream_chat ^2.2.1, stream_chat_flutter 2.2.1 requires stream_chat ^2.2.1.
Because stream_chat 2.2.1 depends on logging ^1.0.1 and no versions of stream_chat match >2.2.1 <3.0.0, stream_chat ^2.2.1 requires logging ^1.0.1.
Thus, stream_chat_flutter 2.2.1 requires logging ^1.0.1.
And because every version of rollbar_flutter depends on logging ^0.11.4, stream_chat_flutter 2.2.1 is incompatible with rollbar_flutter.
So, because roji_wallet depends on both rollbar_flutter 0.1.0-beta and stream_chat_flutter 2.2.1, version solving failed.
pub get failed (1; So, because roji_wallet depends on both rollbar_flutter 0.1.0-beta and stream_chat_flutter 2.2.1, version solving failed.)
exit code 1
```

### PR DESCRIPTION
**Pull request by akornich**
## Description of the change

> adding null-safety
> updating lint tools

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Maintenance
- [ ] New release

## Related issues

> Shortcut stories and GitHub issues (delete irrelevant)

- Fix [SC-90813]
- Fix #10 
- Fix #11

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers assigned
- [x] Issue from task tracker has a link to this pull request
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 local.properties
 **/pana-report
 **/.pubignore
+
+.DS_Store
```
Comment (bishopb) [inline .gitignore:5]: I'd recommend to remove the `.DS_Store` files from the repo, entirely.

**Review by bishopb — DISMISSED** [review]

File: .gitignore (hunk 1 above)
Comment (akornich) [reply .gitignore:5]: @bishopb , agree. missed that one. git ignore probably was not configured for it yet...

File: rollbar_dart/test/client_server_utils.dart [hunk 2]
```diff
@@ -20,42 +20,42 @@ Sender createTextSender(Config c) {
...
+  Future<int?> _start() async {
     _receivePort = ReceivePort();
 
     var tcpInfoPort = ReceivePort();
 
     _isolate = await Isolate.spawn(
         _server, [tcpInfoPort.sendPort, _receivePort.sendPort]);
 
-    int port = await tcpInfoPort.first;
+    var port = await (tcpInfoPort.first);
```
Comment (bxsx) [inline rollbar_dart/test/client_server_utils.dart:39]: Let's skip parentheses, they seem to be redundant

```suggestion
    var port = await tcpInfoPort.first;
```

File: rollbar_dart/test/rollbar_test.dart [hunk 3]
```diff
@@ -35,7 +30,7 @@ void main() {
         failingFunction();
       } catch (error, stackTrace) {
         await rollbar.error(error, stackTrace);
-        var payload = verify(await sender.send(captureAny)).captured.single;
+        var payload = verify(sender.send(captureAny)).captured.single;
```
Comment (bxsx) [inline rollbar_dart/test/rollbar_test.dart:33]: Why don't you want to `await` here? In other test cases, we do await.

```suggestion
        var payload = verify(await sender.send(captureAny)).captured.single;
```

**Review by bxsx — COMMENTED** [review]
Great job @akornich 👍

I leave a few comments.

I also have a question regarding linter settings, however GitHub has a bug and disallows me to do it (similar to: https://github.com/isaacs/github/issues/1338)

> - unawaited_futures
> - unsafe_html

Are these options essential?

I think the `unawaited_futures` could be removed after resolving my comment on rollbar_dart/test/rollbar_test.dart#33

I'm not sure about the `unsafe_html`, but don't see a reason to enable it. Can we ensure HTML is safe and not enable this option?

**Comment by akornich** [comment]
> unawaited_futures

I think the rule is important enough to follow: https://dart-lang.github.io/linter/lints/unawaited_futures.html

**Comment by akornich** [comment]
> unsafe_html

I think the rule makes sense to follow: https://dart-lang.github.io/linter/lints/unsafe_html.html

File: rollbar_flutter/test/platform_transformer_test.dart [hunk 4]
```diff
@@ -1,13 +1,13 @@
 import 'package:flutter_test/flutter_test.dart';
 import 'package:rollbar_dart/rollbar.dart';
-import 'package:rollbar_dart/src/platform.dart';
+//import 'package:rollbar_dart/src/platform.dart';
```
[the remark this answers is not in the document]
Comment (akornich) [reply rollbar_flutter/test/platform_transformer_test.dart]: done. Thanks, @bxsx!

**Comment by akornich** [comment]
@bxsx , thanks for the review! I believe all of your comments are fully resolved...

**Review by bxsx — APPROVED** [review]
Thanks, @akornich for addressing all comments!

And sorry for the confusion with the linter. I misread it. My intention was to keep out unsafe HTML or unawaited futures and it seems the rules do exactly this.

The PR looks good to me now 👍

**Pull request merged**